src/EPR.PRN.Backend.API/Repositories/Repository.cs

```diff
@@ -190,4 +191,58 @@ public async Task<PaginatedResponseDto<PrnDto>> GetSearchPrnsForOrganisation(Gui
         logger.LogInformation("{Logprefix}: Repository - GetSearchPrnsForOrganisation: returning: {DtoObject}", logPrefix, JsonConvert.SerializeObject(dtoObject));
         return dtoObject;
     }
+
+    public async Task SavePrnDetails(Eprn entity)
+    {
+        try
+        {
+            var currentUser = Environment.UserName;
+            var currentTimestamp = DateTime.UtcNow;
+
+            var existingEntity = await _eprContext.Prn.FirstOrDefaultAsync(x => x.PrnNumber == entity.PrnNumber);
+
+            var statusHistory = new PrnStatusHistory
+            {
+                CreatedByOrganisationId = entity.OrganisationId,
+                PrnStatusIdFk = entity.PrnStatusId,
+                CreatedByUser = Guid.Empty,
+                CreatedOn = currentTimestamp,
+            };
+
+            var prnLogVal = entity.PrnNumber?.Replace(Environment.NewLine, "");
+
+            // Add new PRN entity
+            if (existingEntity == null)
+            {
+                entity.CreatedBy = currentUser;
+                entity.CreatedOn = currentTimestamp;
+                _eprContext.Prn.Add(entity);
+                statusHistory.PrnIdFk = entity.Id;
+
+                logger.LogInformation("Attempting to add new Prn entity with PrnNumber : {PrnNumber}", prnLogVal);
+            }
+            // Update existing PRN entity
+            else
+            {
+                entity.Id = existingEntity.Id;
+                _eprContext.Entry(existingEntity).State = EntityState.Modified;
+                _eprContext.Entry(existingEntity).CurrentValues.SetValues(entity);
+
+                statusHistory.PrnIdFk = existingEntity.Id;
+                logger.LogInformation("Attempting to update Prn entity with PrnNumber : {PrnNumber} and {Id}", prnLogVal, entity?.Id);
+            }
+
+            // Add Prn status history
+            _eprContext.PrnStatusHistory.Add(statusHistory);
+
+            await _eprContext.SaveChangesAsync();
+            logger.LogInformation("Prn Entity successfully upserted. PrnNumber : {PrnNumber} and {Id}", prnLogVal, entity?.Id);
+
+        }
+        catch (Exception ex)
+        {
+            logger?.LogError(message: ex.Message, exception: ex);
```

Comment:
why we need to do null check on logger?

Reply:
It complains in Visual Studio if don't do that check